cli/azd/pkg/alphafeatures/alpha_feature_manager.go

```diff
@@ -0,0 +1,91 @@
+package alphafeatures
+
+import (
+	"log"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+)
+
+type AlphaFeatureManager struct {
+	configManager   config.UserConfigManager
+	userConfigCache config.Config
+	// used for mocking alpha features on testing
+	alphaFeaturesResolver func() []AlphaFeature
+}
+
+func NewAlphaFeaturesManager(configManager config.UserConfigManager) *AlphaFeatureManager {
+	return &AlphaFeatureManager{
+		configManager: configManager,
+	}
+}
+
+func (m *AlphaFeatureManager) ListFeatures() (map[string]AlphaFeature, error) {
+	result := make(map[string]AlphaFeature)
+
+	var alphaFeatures []AlphaFeature
+	if m.alphaFeaturesResolver != nil {
+		alphaFeatures = m.alphaFeaturesResolver()
+	} else {
+		alphaFeatures = mustUnmarshalAlphaFeatures()
+	}
+
+	for _, aFeature := range alphaFeatures {
+		// cast is safe here from string to AlphaFeatureId
+		status := disabledText
+		if m.IsEnabled(AlphaFeatureId(aFeature.Id)) {
+			status = enabledText
+		}
+
+		result[aFeature.Id] = AlphaFeature{
+			Id:          aFeature.Id,
+			Description: aFeature.Description,
+			Status:      status,
+		}
+	}
+
+	return result, nil
+}
+
+func (m *AlphaFeatureManager) IsEnabled(featureId AlphaFeatureId) bool {
+	if m.userConfigCache == nil {
+		config, err := m.configManager.Load()
+		if err != nil {
+			log.Panic("Can't load user config!! %w", err)
+		}
+		m.userConfigCache = config
+	}
+
+	//check if all features is ON
+	if allOn := isEnabled(m.userConfigCache, string(AllId)); allOn {
+		return true
+	}
+
+	// check if the feature is ON
+	if featureOn := isEnabled(m.userConfigCache, string(featureId)); featureOn {
+		return true
+	}
+
+	return false
+}
+
+func isEnabled(config config.Config, id string) bool {
+	longKey := strings.Join([]string{
+		string(parentKey),
+		id,
+	}, ".")
+	value, exists := config.Get(longKey)
+	if !exists {
+		return exists
+	}
+	// safe cast -> reading from config text file
+	stringValue, _ := value.(string)
+	stringValue = strings.ToLower(stringValue)
+
+	if stringValue != disabledValue && stringValue != enabledValue {
```

Comment:
I think we have two choices here:

1. We should load all the features and `panic` immediately if a configuration value is invalid, early on in the application
1. Or, we should not `panic` if a feature is misconfigured, and instead we return an `err` that can be formatted nicely in the application

It's not great to have a `panic` later on in the app while doing something.

My preference is number 2

Reply:
I had it like option 2 at one point. I did not like the experience of checking if a feature is enabled or not from. It was like

isEnabled, err := manager.IsEnabled(feature-name)
if err != nil {
   return nil, fmt.error("checking if feature is enabled: %w", err)
}

It feels like the application is expecting that the configuration can be wrong at any moment, so it is worth it to have this as a common error.
I changed it with the intention of make `azd` panic about something is really wrong!! and then, have a difference between errors and panics as:
error: "something went wrong during the normal process of execution."
panic: "something is wrong and the process should not run or continue until it is fixed."

Imagine like an OS where you change some files from the kernel....  OS will not just display an error, it should blue-screen or kernel-panic as the data files are compromised.

Reply:
> UX experience when someone mistypes "on" as "one" 

But this is a side effect from `azd config set foo bar` which applies to everywhere.  The solution there is to show the error about invalid input at that time,,, not after.

The only way the `config` should be invalid is when folks manually update the json file from the user config folder.

Showing a nice error after folks made a mistake (typo) from `azd config` is just cleaning the water comming out of a broken pipe instead of fixing the pipe...

Reply:
I agree with the design consideration, but I think we can make the code slightly less ideal for a better UX experience. We can refactor the code later.